app/models/Notification.scala

```diff
@@ -17,12 +17,65 @@
 package models
 
 import org.joda.time.{DateTime, DateTimeZone}
-import play.api.libs.json.{Format, Json}
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import reactivemongo.bson.BSONObjectID
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
-case class Notification(fileReference: String, outcome: String, filename: String, createdAt: DateTime = DateTime.now.withZone(DateTimeZone.UTC))
+case class NotificationDetails(fileReference: String, outcome: String, filename: String)
+
+object NotificationDetails {
+  implicit val writes: OWrites[NotificationDetails] = Json.writes[NotificationDetails]
+  implicit val reads: Reads[NotificationDetails] =
+    ((__ \ "fileReference").read[String] and
+      (__ \ "outcome").read[String] and
+      (__ \ "filename").read[String])(NotificationDetails.apply _)
+
+  implicit val format: Format[NotificationDetails] = Format(reads, writes)
+}
+
+case class Notification(
+  _id: BSONObjectID,
+  payload: String,
+  details: Option[NotificationDetails] = None,
+  createdAt: DateTime = DateTime.now.withZone(DateTimeZone.UTC)
+)
 
 object Notification {
-  implicit val dateFormat: Format[DateTime] = ReactiveMongoFormats.dateTimeFormats
-  implicit val notificationFormat = Json.format[Notification]
+  object DbFormat extends ReactiveMongoFormats {
+    val writes: Writes[Notification] =
+      ((JsPath \ "_id").write[BSONObjectID] and
+        (JsPath \ "payload").write[String] and
+        (JsPath \ "details").writeNullable[NotificationDetails](NotificationDetails.writes) and
+        (JsPath \ "createdAt").write[DateTime])(unlift(Notification.unapply))
+
+    implicit val reads: Reads[Notification] =
+      ((__ \ "_id").read[BSONObjectID] and
+        (__ \ "payload").read[String] and
+        (__ \ "details").readNullable[NotificationDetails] and
+        (__ \ "createdAt").read[DateTime])(Notification.apply _)
```

Comment:
Nit: Could these be possibly code-space-optimized by simply using `Json.format[Notification]` ?

Reply:
:+1: